app/controllers/api/v0/tags_controller.rb

```diff
@@ -1,20 +1,24 @@
 module Api
   module V0
     class TagsController < ApplicationController
-      caches_action :index,
-                    cache_path: proc { |c| c.params.permit! },
-                    expires_in: 10.minutes
-      before_action :set_cache_control_headers, only: %i[onboarding] # essentially static content
       respond_to :json
 
+      before_action :set_cache_control_headers, only: %i[index onboarding]
+
       def index
-        @page = params[:page]
-        @tags = Tag.order(taggings_count: :desc).page(@page).per(10)
+        page = params[:page]
+        per_page = (params[:per_page] || 10).to_i
+        num = [per_page, 1000].min
+
+        @tags = Tag.order(taggings_count: :desc).page(page).per(num)
+
+        set_surrogate_key_header Tag.table_key, @tags.map(&:record_key)
```

Comment:
Since this is cached with the table_key do we need to do anything special to purge this table_key when a Tag is updated or added or deleted? I noticed we do `tag.purge` but wasn't clear if that will include the table_key as well.

Reply:
in actuality `tag.purge` won't purge _all tags_, only the cache entries which contain the tag being purged but yes, it's good enough for now. Once we'll have a complete overview of all caching and purging we can start employing `.soft_purge_all` and `.purge_all` where needed